validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -308,8 +308,8 @@ def _squash(state_root, context_ids):
         return _squash
 
     def stop(self):
-        self._context_writer.join(1)
-        self._context_reader.join(1)
+        self._address_queue.put_nowait(-1)
```

Comment:
Declaring a module-level variable for -1 and using it here and below would make it clearer.